src/Symfony/Component/Security/Core/Authorization/Voter/Vote.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+/**
+ * A Vote is returned by a Voter and contains the access (granted, abstain or denied).
+ * It can also contain a reason explaining the vote decision.
+ *
+ * @author Dany Maillard <danymaillard93b@gmail.com>
+ */
+final class Vote
+{
+    use AccessTrait;
+
+    private $reason;
+    private $parameters;
+
+    /**
+     * @param int $access One of the VoterInterface::ACCESS_* constants
+     */
+    private function __construct(int $access, string $reason = '', array $parameters = [])
+    {
+        $this->access = $access;
+        $this->reason = $reason;
+        $this->parameters = $parameters;
+    }
+
+    public static function create(int $access, string $reason = '', array $parameters = []): self
```

Comment:
What is the benefit of this compared to the constructor ?

Reply:
I'm not sure. The original contributor is using private for __construct. I'm unsure why he/she made that decision though. I can remove this method then update code everywhere? My guess is that he/she wants all methods in Vote to be static.

Reply:
> My guess is that he/she wants all methods in Vote to be static.

And that's quite the opposite of existing Symfony APIs. In some version (I don't remember which one, check the changelogof HttpFoundation if you care about it), we deprecated `Response::create()` and similar methods that were duplicating the constructor signature.